### ISSUE TITLE
KB randomly stops receiving power, reporting status "Not charging" and -0.1A until replugged

### ISSUE THREAD

**Issue by lily-bit-0** [estokes/pinephone-kbpwrd#3]
Reoccurringly while charging, kbpwrd suddenly reports the KB status as full and stops charging it, never recovering. But it’s not actually full according to manual and kbpwrd’s readings of OCV and capacity, and neither is its status set to full according to manual readings. Replugging the charger makes it work again.

**Comment by lily-bit-0** [comment]
Like #4, this also happens when using my own new Bash script to automatically set limits. What I didn’t know when I made this bug report is that the actual status in ´status´ reports ´Not charging´ instead of ´Full´ as kbpwrd says. Apparently in your code you set ´Not charging´ to be interpreted like ´Full´, but at least for me right now this state does not equal full and it only occurs when the KB is off without a power supply or bugged. This bug differs from the two bugs in #4 because it still reports all values while not charging from power supplies and always displays a KB current of -0.1A (#2). How many times the script loops or writes to values seems to make no difference, it appears to have the same chance of happening when reading and writing values every second as when reading once a minute and only writing when the new value would be different. Strangely enough this not only happens when my limit script or kbpwrd is running but now also when my old and much used script that prints battery values in an interval is running while charging. It never caused this before. The only thing I changed was adding current limits, status and online to what it reads and prints. Charging without running kbpwrd, my automatic limit script or my monitoring script does not trigger this bug, neither does changing the limits manually for PP charging, KB charging and discharging, or my monitoring script until the last change. So far I’ve been unable to figure out what exactly causes this or how I could work around it. Since this bug stops charging until the power supply is replugged, I can’t have my script or kbpwrd running while charging. I guess for now I have to manually call my script (via a global hotkey for at least some convenience) once with no loops whenever I plug in a charger and again when the PP is full, at least when I know the PP would otherwise go down to 0%. That hasn’t caused this bug the few times I did it, I guess and hope the chance of it happening immediately like this is very small. Usually it takes a couple of minutes up to an hour of kbpwrd or my script running for this to occur.
